### DISCUSSION TITLE
Take Leaf off-line temporarily for maintenance?

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#446] [Q&A] [answered]
Hi Leaf Folks

Is there a convenient way to take Leaf off-line temporarily for maintenance? Our clinical data warehouse will be updated daily. Following these updates, we will run scripts that rewrite the Concepts in the Leaf DB and recalculate the data that depends on them, such as patient counts and concept indices. To avoid showing incomplete or inconsistent data, Leaf must be taken off-line while the Leaf DB is being updated.

Perhaps this feature already exists in Leaf.

If not, ideally during maintenance users could still login to Leaf but would see a notice that "Leaf is down for maintenance until <an end date and time>". In addition, Leaf would warn users of upcoming maintenance.

My preference would be to control Leaf's maintenance state via the LeafDB, so it can be managed by database updates, for example by the scripts that will update the Leaf DB daily.

Thanks
Arthur

**Comment by ndobb** [comment, marked as answer]
Hi @artgoldberg, thanks for your patience in our reply.

This feature doesn't exist quite yet. The "traditional" way we've done this in the past is to swap the production `index.html` file with another simple page with wording similar to what you have above, then swap back when the DB is good-to-go again. This isn't necessarily the best way to do it, but we've not baked in any features to effectively keep the app online but essentially prevent logins.

It sounds like it may be convenient though, particularly if admins could still get in as well as customize the "down for maintenance until..." message. 

Could you please make an issue for this? A short blurb just referencing this conversation would be great.

Thanks,
-nic

**Reply by artgoldberg** [reply]
Hi @ndobb 

You’re welcome.

Thanks for considering my request. 

Yes, your “traditional” approach works for people who have access to the app server and can change its index.html. As you point out, a drawback of this approach is that it makes Leaf inaccessible to admins as well as regular users, can make Leaf look different to users when the “unavailable” home page is present, and would be inconvenient in a scenario when Leaf was frequently – e.g. nightly -- taken off-line for maintenance.

I will create an issue as you requested.

Regards
Arthur

**Reply by artgoldberg** [reply]
Moved to Leaf issue #453
